impl/client/src/main/doc-js/pentaho/type/spec/ITypeDescriptor.jsdoc

```diff
@@ -0,0 +1,48 @@
+/*!
+ * Copyright 2018 Hitachi Vantara. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * The `spec.ITypeDescriptor` interface describes a type descriptor in _object_ form.
```

Comment:
Why was the TypeDescriptor needed? I remember vaguely... It had to do with the data table types vs. the type api types?

Reply:
`TypeDescriptor` is `Simple` type which wraps a `pentaho.type.Type` instance. It allows a property of a complex type to hold a type. The use case is that of `pentaho.visual.role.Mode#dataType`. Then `pentaho.visual.role.Property.Type#modes` — a type attribute — holds a list of `Mode` instances. Up until now, we didn't have a use case like this; types always referenced other types directly and not through a complex value.